src/appleseed/foundation/math/fresnel.h

```diff
@@ -252,6 +254,20 @@ void fresnel_dielectric_schlick(
     reflectance += SpectrumType(static_cast<ValueType>(k5 * multiplier));
 }
 
+template <typename T>
+T fresnel_transmission(const T cos_theta, const T eta)
+{
+    const T sin_theta_t_sqr = T(1.0) / square(eta) * (T(1.0) - square(cos_theta));
```

Comment:
Assuming there are no missing parentheses,

    const T sin_theta_t_sqr = (T(1.0) - square(cos_theta)) / square(eta);

may be faster.